### ISSUE TITLE
Extend K8s tool to add api headers i.e User impersonation 

### ISSUE THREAD

**Issue by GinoSahagun** [vscode-kubernetes-tools/vscode-kubernetes-tools#1052]
Hi team,

I couldn't find anything related to my inquiry in the issues, but I wanted to know if there is something on the road map to add kubectl api headers to extend kubectl commands through the vscode tool. Our team sometimes need to do something like `kubectl --as <name-of-elevator-user> get secrets` depending on the kubectl command and I thought it'd be really cool if the vscode tool could add some additional options to add certain headers to make the commands similar to the manual process.

I apologize if I didn't word this very well. this is my first time writing a post like this.  I'm happy to contribute if this at all is reasonable.

**Comment by lstocchi** [comment]
Hey @GinoSahagun thanks for opening this. We would need to update the code a bit but i think it could be a nice enhancement. How do you think this should work? Having an action similar to the incognito mode in the browser so that when switched all commands are executed by impersonating another user/group?

 @itowlson @squillace WDYT?

**Comment by GinoSahagun** [comment]
Yes...  I like that suggestion it makes sense and it seems to align how kubernetes describes the "impersonation mode" and how users / groups can be used.

**Comment by itowlson** [comment]
Sorry for the belated reply.  Impersonation mode sounds like the most practical way in the current design; the only other option I can think of is to have different tree entries you could switch between ("My Cluster" / "My Cluster As Seen By Keith").

This seems like it could have the potential for confusion if someone does non-k8s stuff for a while and forgets they're in impersonation mode.  What UI can/should we provide to remind them?  Browsers tend to change their chrome for private mode but we can't really mess with VS Code that way...!

**Comment by GinoSahagun** [comment]
Sorry for my belated reply. is it possible to have the present being similar to how clusters are shown, but have the user be shown in a similar structure and using a tab maybe or something?
